datafusion/core/tests/memory_limit/mod.rs

```diff
@@ -323,6 +323,31 @@ async fn oom_recursive_cte() {
         .await
 }
 
+#[tokio::test]
+async fn oom_parquet_sink() {
+    let file = tempfile::Builder::new()
+        .suffix(".parquet")
+        .tempfile()
+        .unwrap();
+
+    TestCase::new()
+        .with_query(format!(
+            "
+            COPY (select * from t)
+            TO '{}'
+            STORED AS PARQUET OPTIONS (compression 'uncompressed');
+        ",
+            file.path().to_string_lossy()
+        ))
+        .with_expected_errors(vec![
+            // TODO: update error handling in ParquetSink
+            "Unable to send array to writer!",
```

Comment:
The parallelized writes have vectors for channels, and vectors for the spawned tasks. This error we are hitting (on memory limit reached) is for the closed channel.

I believe we want to be surfacing errors for the tasks, which **should** exit due to the memory reservation error. Need to poke around a bit more.

Reply:
I think we need to update several map_err statements to propagate inner error messages rather than ignore them. E.g. 

https://github.com/apache/datafusion/blob/b96186fdef1ff410663ec8fce41186c018f8e09a/datafusion/core/src/datasource/file_format/parquet.rs#L880-L884

change to something like

```rust
            col_array_channels[next_channel]
                .send(c)
                .await
                .map_err(|e| internal_datafusion_err!("Unable to send array to writer due to error {e}"))
```

Reply:
Filed https://github.com/apache/datafusion/issues/11397 to track